app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -510,19 +508,14 @@ class LuciBuildService {
     return false;
   }
 
-  bool _shouldRerunBuilder(LuciTask luciTask, String repo) {
-    if (luciTask.summaryMarkdown == null || !luciTask.builderName.contains('Mac')) {
+  /// Only blocking failures on framework TOT will be triggered.
+  Future<bool> _shouldRerunBuilder(LuciTask luciTask, int retries, Commit commit, DatastoreService datastore) async {
+    if ((luciTask.status != Task.statusFailed && luciTask.status != Task.statusInfraFailure) ||
+        retries >= config.maxLuciTaskRetries ||
+        !Config.schedulerSupportedRepos.contains(commit.slug)) {
       return false;
     }
-    switch (repo) {
-      case 'flutter':
-        return luciTask.summaryMarkdown.contains('retcode: -9') ||
-            (kMacBuildersWithShards.contains(luciTask.builderName) &&
-                luciTask.summaryMarkdown ==
-                    'recipe infra failure: Infra Failure: Step(\'display builds.build(s) failed\') (retcode: 1)');
-      case 'engine':
-        return luciTask.summaryMarkdown.contains('retcode: -9');
-    }
-    return false;
+    final Commit latestCommit = await datastore.queryRecentCommits(limit: 1).single;
```

Comment:
How do we prevent this from running on almost every failed task of every commit?

Reply:
If the commit is TOT, it is expected that all failed tasks will be rerun.
If the commit is not TOT, then failed tasks will not be triggered.

Reply:
Let's keep an eye on the behavior once it lands, given that the cron job is not in sync with the scheduler it may cause unexpected race conditions potentially leading to retry almost all failures.

Reply:
We have deprecated the cron job and solely rely on the scheduler now, so we should be okay in this area.
But I will closely monitor the dashboard on Monday.